framework/src/preconditioners/FieldSplitPreconditioner.C

```diff
@@ -72,7 +72,7 @@ FieldSplitPreconditioner::FieldSplitPreconditioner(const InputParameters & param
       for (unsigned int j = 0; j < n_vars; j++)
         (*cm)(i,j) = 1; // full coupling
   }
-  _fe_problem.setCouplingMatrix(cm);
+  _fe_problem.setCouplingMatrix(std::move(cm));
```

Comment:
The new method takes a parameter of type `std::unique_ptr<CouplingMatrix>` so it'll do the right thing. The old pointer based method is deprecated and will be removed after some of the downstream apps are fully patched.